torch/csrc/CudaIPCTypes.cpp

```diff
@@ -25,13 +25,20 @@ void warnProducerTerminatedBeforeSharedTensorsReleased() {
 }
 
 struct CudaIPCGlobalEntities {
+  // This class is used as a singleton (see cuda_ipc_global_entities)
+  // This variable is used to track its lifetime to avoid accessing it
+  // after it was destroyed which would lead to segmentation faults
+  // Note that a trvial type is used which doesn't suffer from construction
+  // and destruction order issues
+  static bool alive;
+
   std::mutex ref_counters_mutex_;
   std::atomic<int64_t> sync_events_used_;
   std::map<std::string, std::shared_ptr<CudaIPCRefCountersFile>>
       ref_counters_files_;
   std::shared_ptr<CudaIPCRefCountersFile> next_available_ref_counters_file_;
   CudaIPCSentDataLimbo CudaIPCSentDataLimbo_;
-  CudaIPCGlobalEntities() : ref_counters_files_() {}
+  CudaIPCGlobalEntities() { alive = true; }
```

Comment:
nit: maybe leave ref_counters_files_() call?

Reply:
I did remove it for consistency while I'm already touching that line. `ref_counters_files_` is a map, which has a default ctor, same as mutex and shared_ptr, so if you call its default ctor explicitly then I'd guess the other should be as well, but it doesn't change anything as C++ default constructs those already, so IMO less code is better